xtractmime/_utils.py

```diff
@@ -256,17 +290,49 @@ def is_audio_video(input_bytes: bytes) -> Optional[bytes]:
     return None
 
 
-def is_font(input_bytes: bytes) -> Optional[bytes]:
+def get_font_mime(input_bytes: bytes) -> Optional[bytes]:
     for pattern in FONT_PATTERNS:
-        if _is_match_mime_pattern(input_bytes, pattern[0], pattern[1], pattern[2]):
+        if is_match_mime_pattern(input_bytes, pattern[0], pattern[1], pattern[2]):
             return pattern[3]
 
     return None
 
 
-def is_archive(input_bytes: bytes) -> Optional[bytes]:
+def get_archive_mime(input_bytes: bytes) -> Optional[bytes]:
     for pattern in ARCHIVE_PATTERNS:
-        if _is_match_mime_pattern(input_bytes, pattern[0], pattern[1], pattern[2]):
+        if is_match_mime_pattern(input_bytes, pattern[0], pattern[1], pattern[2]):
             return pattern[3]
 
     return None
+
+
+def get_text_mime(input_bytes: bytes) -> Optional[bytes]:
+    for pattern in TEXT_PATTERNS:
+        if is_match_mime_pattern(input_bytes, pattern[0], pattern[1], pattern[2]):
+            return pattern[3]
+
+    return None
+
+
+def get_extra_mime(
+    input_bytes: bytes,
+    extra_types: Optional[Tuple[Tuple[bytes, bytes, Optional[Set[bytes]], bytes], ...]],
+) -> Optional[bytes]:
+    for pattern in EXTRA_PATTERNS:
+        if is_match_mime_pattern(input_bytes, pattern[0], pattern[1], pattern[2]):
+            return pattern[3]
+
+    if extra_types:
+        for pattern in extra_types:
```

Comment:
The reason `mypy` complains here is that in line 321 `mypy` automatically determines the signature of `pattern` to be the signature of each item of `EXTRA_PATTERNS`, none of which has `None` as the 3rd tuple item, so it does not expect that item to allow None values.

Because here, in line 326, you use again the `pattern` variable, `mypy` expects the variable to remain having the same type that `mypy` guessed in line 321.

The best solution, I think, would be to edit `_patterns.py` and set the right type hint for `EXTRA_PATTERNS`, so that it matches the signature of the `extra_types` parameter here:

```python
EXTRA_PATTERNS: Tuple[Tuple[bytes, bytes, Optional[Set[bytes]], bytes], ...] = (…)
```

Reply:
Fixed it,
Thankyou